doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -1227,3 +1227,39 @@ Exposing constants to OCaml libraries
 
    Due to its internal nature, this command is not for general use. It is meant
    to appear only in standard libraries and in support libraries of plug-ins.
+
+   As a special case, when the first segment of :n:`@qualid__2` is :g:`kernel`,
+   the constant is exposed to the kernel. For instance, the `Int63` module
+   features the following declaration:
+
+   .. coqdoc::
+
+      Register bool as kernel.ind_bool.
+
+   This makes the kernel aware of what is the type of boolean values. This
+   information is used for instance to define the return type of the
+   :g:`#int63_eq` primitive.
+
+Inlining hints for the fast reduction machines
+----------------------------------------------------------------
+
+.. cmd:: Register Inline @qualid
+
+   This command gives as a hint to the reduction machines (VM and native) that
+   the body of the constant :n:`@qualid` should be inlined in the generated code.
+
+   Due to its internal nature, this command is not for general use. It is meant
+   to appear only in standard libraries and in support libraries of plug-ins.
+
+Registering primitive operations
+--------------------------------
+
+.. cmd:: Primitive @ident__1 := #@ident__2.
+
+   Declares :n:`@ident__1` as the primitive operator :n:`#@ident__2`. When
+   running this command, the type of the primitive should be already known by
+   the kernel (this is achieved through this command for primitive types and
+   through the :cmd:`Register` command with the :g:`kernel` name-space).
```

Comment:
Did you mean something like this?

```suggestion
   through the :cmd:`Register` command with the :g:`kernel` name-space for functions).
```

Reply:
What do you call a function? We are talking about types (`int`, `bool`, `comparison`…) some of which are primitive, some of which are not.

Reply:
Updated. Thanks.